pydatalab/pydatalab/apps/xrd/blocks.py

```diff
@@ -39,11 +39,35 @@ def load_pattern(
         if ext == ".xrdml":
             df = parse_xrdml(location)
 
-        elif ext == ".xy":
-            df = pd.read_csv(location, sep=r"\s+", names=["twotheta", "intensity"])
-
         else:
-            df = pd.read_csv(location, sep=r"\s+", names=["twotheta", "intensity", "error"])
+            if ext == ".xye":
+                columns = ["twotheta", "intensity", "error"]
+            else:
+                columns = ["twotheta", "intensity"]
```

Comment:
This will cause a problem with the I11 data files from previous beamlines, which have ".dat" extension but are 3 column xye. Though overall this gets messy because other data manufacturers use .dat to mean totally different formats unfortuantely

Reply:
quick fix is just to do `if ext in (".xye", ".dat")`, which I guess should at least work with the file types we are currently working with

Reply:
Another option is to try detect how many columns there are if the filetype is not explicit

Reply:
But thanks for the heads up about .dat...

Reply:
Actually this is straightforward, we can just always try to read as 3 columns and the `error`s will be set to NaN if missing

Reply:
(right now we don't use the errors anywhere, but once we have a schema  we can just `null` out any NaNs)